expan/core/experiment.py

```diff
@@ -383,33 +330,36 @@ def delta(self, method='fixed_horizon', kpi_subset=None, derived_kpis=None, **kw
 				# store the reference metric name to be used in the weighting
 				# TODO: only works for ratios
 				res.metadata['reference_kpi'][kpiName] = re.sub(pattern+'/', '', dk['formula'])
+				reference_kpis[kpiName] = re.sub(pattern+'/', '', dk['formula'])
 
 		if kpi_subset is not None:
 			kpis_to_analyse.intersection_update(kpi_subset)
 		self.dbg(3, 'kpis_to_analyse: ' + ','.join(kpis_to_analyse))
 
+		defaultArgs = [res, kpis_to_analyse]
+		deltaWorker = statx.make_delta(assume_normal, percentiles, min_observations,
+				                       nruns, relative)
 		method_table = {
-			'fixed_horizon':    self.fixed_horizon_delta,
-			'group_sequential': self.group_sequential_delta,
-			'bayes_factor':     self.bayes_factor_delta,
-			'bayes_precision':  self.bayes_precision_delta,
+			'fixed_horizon':    (self.fixed_horizon_delta,    defaultArgs + [reference_kpis, weighted_kpis]),
+			'group_sequential': (self.group_sequential_delta, defaultArgs                                  ),
```

Comment:
The thing is the following: most old `delta()`-like methods (`fixed_horizon_delta()`, `sga()`, `trend()`, `feature_check()`, etc., use the same low-level `delta()` backend and thus share its common numerical parameters. That's why it seems to be a good idea to pass a partially applied `delta()` around (in the form of `deltaWorker`) and thus spare the number of arguments.

All new Bayesian funcs have a separate low-level statistical worker. In that sense, the umbrella func `delta()` relies on the default parameters of those and passing a partially applied worker would not make much sense.

We have to think about decoupling the old `*_delta()` code from the new one.